components/tikv_util/src/range_latch.rs

```diff
@@ -95,66 +95,55 @@ impl RangeLatch {
                 );
                 debug_assert!(previous_value.is_none());
 
-                // Now acquire the lock after releasing the write guard
-                let lock = mutex.lock().unwrap();
-                // The lifetime of the lock returned by `mutex.lock()` is tied to the `_guard`
-                // variable, but we need to extend its lifetime to `'static` so it can be stored
-                // in the `RangeLatchGuard` struct. This is safe because:
-                // - The `lock` is managed by the `RangeLatchGuard`, ensuring proper drop
-                //   semantics.
-                // - The lifetime extension does not result in use-after-free, as the lock's
-                //   underlying resources are valid for the lifetime of the `mutex`.
-                // - The design guarantees that `RangeLatchGuard` will hold the lock for its
-                //   entire lifetime, preventing invalid access.
-                let lock = unsafe { std::mem::transmute(lock) };
+                // Now acquire the latch after releasing the write guard
+                let mutex_guard = mutex.lock().unwrap();
+                // The `mutex_guard` is being extended to `'static` because its
+                // default lifetime is tied to the scope of this function (`'a`), as determined
+                // by Rust's borrow checker. By extending the lifetime to `'static`, we allow
```

Comment:
you can add a lifetime for `RangeLatchGuard<'a>`, so no need to make the lifetime `'static`